### PR TITLE
Integrate ovsdb-etcd to ovn-heater and create etcd test scenario.

### PR DESCRIPTION
**Pull request by msherif1234**
Signed-off-by: Mohamed S. Mahmoud <mmahmoud@redhat.com>

### CODE REVIEW

File: ovn-tester/ovn_tester.py [hunk 1]
```diff
@@ -114,7 +114,9 @@ def read_config(config):
         vips=cluster_args.get('vips', calculate_default_vips()),
         vip_subnet=DEFAULT_VIP_SUBNET,
         static_vips=cluster_args.get('static_vips',
-                                     calculate_default_static_vips())
+                                     calculate_default_static_vips()),
+        use_ovsdb_etcd=cluster_args.get('use_ovsdb_etcd', False),
+        enable_ssl=cluster_args.get('enable_ssl', False)
```
Comment (dceara) [inline ovn-tester/ovn_tester.py:119]: SSL used to be enabled by default, let's leave it like that.

File: physical-deployments/physical-deployment.yml [hunk 2]
```diff
@@ -1,9 +1,8 @@
-registry-node: host01.mydomain.com
-internal-iface: eno1
+registry-node: localhost
```
Comment (dceara) [inline physical-deployments/physical-deployment.yml:1]: I guess the physical-deployment.yml change was committed by accident.

File: test-scenarios/ovn-etcd-low-scale.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+global:
+  log_cmds: True
```
Comment (dceara) [inline test-scenarios/ovn-etcd-low-scale.yml:2]: Probably better to make it less verbose by disabling command logging.

File: test-scenarios/ovn-low-scale.yml [hunk 4]
```diff
@@ -3,6 +3,7 @@ global:
 
 cluster:
   clustered_db: true
+  enable_ssl: False
```
Comment (dceara) [inline test-scenarios/ovn-low-scale.yml:6]: This seems unrelated, please revert this.

**Review by dceara — CHANGES_REQUESTED** [review]
Thanks for the contribution, I left a few comments.  Also, it would be great to mention what dependencies (e.g., ovn-fake-multinode, ovsdb-etcd) this has.

File: test-scenarios/ovn-low-scale.yml (hunk 4 above)
Comment (msherif1234) [reply test-scenarios/ovn-low-scale.yml:6]: > This seems unrelated, please revert this.

its very related for ovs-etcd we need to use tcp protocol since the client only listen to tcp on port 6641

File: test-scenarios/ovn-etcd-low-scale.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+global:
+  log_cmds: False
+
+cluster:
+  cluster_cmd_path: /home/mmahmoud/ovn-heater/runtime/ovn-fake-multinode
```
Comment (msherif1234) [inline test-scenarios/ovn-etcd-low-scale.yml:5]: @dceara I need the above setting w/o containers won't start and in the same time I can't leave like that what are you suggestions ?

Comment (dceara) [reply test-scenarios/ovn-etcd-low-scale.yml:5]: @msherif1234 Sure, but that can be a local change, in your repo.  These files are "samples" of test scenarios and should probably be as generic as possible.  Ideally we should get rid of `cluster_cmd_path` completely but that's a longer story and is not relevant for this PR.

**Comment by dceara** [comment]
@msherif1234 There are also a few failed [flake8 checks](https://github.com/dceara/ovn-heater/runs/3173914118?check_suite_focus=true), could you please fix those too?

File: ovn-tester/ovn_workload.py [hunk 6]
```diff
@@ -117,10 +122,15 @@ def start(self, cluster_cfg):
     @ovn_stats.timeit
     def connect(self, cluster_cfg):
         print(f'***** connecting worker {self.container} *****')
+        if not cluster_cfg.enable_ssl:
+            node_remote = \
+                'tcp:192.16.0.1:6642,tcp:192.16.0.2:6642,tcp:192.16.0.3:6642'
```
Comment (dceara) [inline ovn-tester/ovn_workload.py:127]: I think the best way to do this is to actually change calculate_default_node_remotes in ovn_tester.py and pass it enable_ssl based on which it should either set tcp or ssl as protocol [here](https://github.com/dceara/ovn-heater/blob/6aeafc0ec64296cf7f018700fed9df8904fa482d/ovn-tester/ovn_tester.py#L59).

**Review by dceara — CHANGES_REQUESTED** [review]

**Review by dceara — APPROVED** [review]
Looks good to me, thanks! I'm waiting for the [ovn-fake-multinode PR](https://github.com/ovn-org/ovn-fake-multinode/pull/46) to merge first before pushing this one.
